static/js/src/ua-payment-methods.js

```diff
@@ -89,15 +104,152 @@ updateButton.addEventListener("click", function () {
       card: card,
     })
     .then((result) => {
-      if (result.paymentMethod) {
-        setPaymentMethod(window.accountId, result.paymentMethod.id).then(() => {
-          location.reload();
-        });
-      } else {
-        handleError(result.error);
+      if (!result.paymentMethod) {
+        handleError();
+        return;
       }
+
+      handlePaymentMethod(result);
     })
-    .catch((error) => {
-      handleError(error);
+    .catch(() => {
+      handleError();
     });
 });
+
+if (cancelButton)
+  cancelButton.addEventListener("click", () => {
+    previewSection.classList.remove("u-hide");
+    editSection.classList.add("u-hide");
+    paymentErrorElement.classList.add("u-hide");
+    paymentErrorElement.querySelector(".p-notification__message").innerHTML =
+      "";
+  });
+
+const handleSuccess = (message) => {
+  paymentErrorElement.classList.add("u-hide");
+  paymentWarningElement.classList.add("u-hide");
+  paymentSuccessElement.classList.remove("u-hide");
+  paymentSuccessElement.querySelector(".p-notification__message").innerHTML =
+    message + ". Reloading page...";
+};
+
+const handleError = () => {
+  updateButton.classList.remove("is-processing");
+  updateButton.innerHTML = "Update";
+  updateButton.disabled = false;
+  if (cancelButton) cancelButton.disabled = false;
+};
+
+const handlePaymentMethodErrors = (message = null) => {
+  if (!message) {
+    message = "Sorry, there was an error with the payment.";
```

Comment:
Remove the Sorry as Ant suggested

```suggestion
    message = "There was an error with the payment.";
```

Reply:
Changed!